src/gmp/z.d

```diff
@@ -1081,6 +1093,23 @@ private struct _MpZ(bool copyable = false)
         }
     }
 
+    /** Exports to arbitrary words of binary data into `rop`. It's format defined by:
+        - `order`: the most significant word `first` or `last` for least significant first
+        - `size` in bytes of each word
+        - `endian` can be `bigEndian`, `littleEndian` or `host` default
+        - the most significant `nails` bits of each word are unused and set to zero, this can be 0 to produce full words
+
+        Returns: the number of words produced
+     */
+    pragma(inline, true)
+    size_t convert(void* rop, WordOrder order, size_t size, Endianess endian, size_t nails) const @trusted
```

Comment:
We should strive for replacing `rop` and return value with a returned D array (slice) of `void`, `ubyte`, `size_t` or `uint` instead of a `void*`.

If you don't want the store to be GC-allocated you should at least pass the preallocated store as a slice instead. In debug mode you should check that size of the pre-allocated array is at least as large as `mpz_sizeinbase`. See https://gmplib.org/manual/Integer-Import-and-Export.html for details.

Reply:
> We should strive for replacing rop and return value with a returned D array (slice) of void, ubyte, size_t or uint instead of a void*.

Done. 

> If you don't want the store to be GC-allocated you should at least pass the preallocated store as a D slice (scope T[] rop) instead. In debug mode you should check that size of the pre-allocated array (T.sizeof*rop.length) is at least as large as mpz_sizeinbase. See https://gmplib.org/manual/Integer-Import-and-Export.html for details. The slice returning variant of convert should call this non-allocating convert.

Hm, what will be signarure of the method, then? I.e. non-allocating array will be as the current one (`rop, WordOrder order, size_t size, Endianess endian, size_t nails`), and the allocating array should not take the `rop`, allocate it inside and return tuple of new allocated slice + number of words?